high_card.rb

```diff
@@ -1,34 +1,42 @@
-def build_deck
-  values = [2, 3, 4, 5, 6, 7, 8, 9, 10, 'J', 'Q', 'K', 'A']
-  suits = ['hearts', 'diamonds', 'clubs', 'spades']
-  deck = []
+suits = ["spades", "clubs", "hearts", "diamonds"]
+ranks = [2,3,4,5,6,7,8,9,10, "J", "Q", "K", "A"]
+deck = ranks.product(suits).shuffle
+# deck = []
+# ranks.each do |rank|
+#   suits.each do |suit|
+#     deck << [rank, suit]
+#   end
+# end
+players = []
+puts players.length.to_s + "Players so Far"
+puts "Enter a player name or type play: "
+player = gets.chomp
+while player != "play" do
+    players << player
+    puts "Enter a player name or type play: "
+    player = gets.chomp
+end
 
-  values.each_with_index do |v, i|
-    suits.each do |s|
-      deck.push({
-        score: i,
-        value: v,
-        suit: s,
-      })
-    end
-  end
+players_with_cards = players.map do |player|
+  {name: player, card: nil}
+end
 
-  return deck.shuffle
+players_with_cards.each do |player|
+  card = deck.pop
+  player[:card] = card
 end
+puts players_with_cards.inspect
```

Comment:
This is a great technique for debugging, and figuring out what's going on! 👍  Careful about leaving puts statements after you figure out what's happening.